packages/x-date-pickers/src/AdapterMoment/AdapterMoment.ts

```diff
@@ -145,23 +145,6 @@ export class AdapterMoment implements MuiPickersAdapter<Moment, string> {
     return value.locale(expectedLocale);
   };
 
-  /**
-   * Some methods from moment can't take the locale as a parameter and always use the current locale.
-   * To respect the adapter locale, we need to set it as the current locale and then reset the previous locale.
-   */
-  private syncMomentLocale = <R>(runner: () => R): R => {
```

Comment:
That's a nice side effect of the new  `getWeekdays`, we don't use the moment method that did not accept the locale as a param.